{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "6a935406-25f9-4bc4-be60-5078943a3daf",
   "metadata": {},
   "source": [
    "# Match sample metadata to NOAA sheet\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "id": "9530a10d-2360-42fb-8cac-3c0ceda1f1ef",
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import numpy as np\n",
    "import datetime\n",
    "import matplotlib.pyplot as plt\n",
    "import glob\n",
    "\n",
    "#For illustrator import:\n",
    "plt.rcParams['pdf.fonttype'] = 42\n",
    "plt.rcParams['ps.fonttype'] = 42"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "95691426-a366-4beb-ba75-22bfac2f3d61",
   "metadata": {},
   "source": [
    "### Import Dada2 data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "id": "d0c6ddd6-5002-41cd-a577-791be0f70ab7",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "XXXXXX     12S\n",
      "Number samples in otu_table: 286\n",
      "['FilterID', 'target_gene', 'PlateID', 'library', 'local_time', 'time_label', 'SAMPLING_cruise', 'depth', 'SAMPLING_platform', 'SC', 'ESP', 'SAMPLING_station_number', 'SAMPLING_station', 'SAMPLING_bottle', 'decimalLongitude', 'decimalLatitude', 'sample_type', 'Plates', 'Markers', 'Status', 'Dewar_name', 'Sampling_method', 'replicate', 'SAMPLING_rdepth', 'project_name', 'nitrate', 'fluor', 'density', 'pressure', 'minimumDepthInMeters', 'maximumDepthInMeters', 'start_GMT', 'end_GMT', 'temp', 'salinity', 'sigmat', 'spice', 'diss_oxygen', 'PAR (umol/s/m2)', 'altitude', 'chlorophyll', 'bbp470 (count)', 'bbp650 (count)', 'SAMPLING_project', 'ESP_name', 'diel', 'month', 'day', 'hour', 'eventDate']\n",
      "Number samples in metadata table: 286\n",
      "Number ASVs in taxa table: 2827\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/var/folders/5v/h4s_0pz17yb6f6krsxkzr5dw000b65/T/ipykernel_84247/4191855385.py:19: UserWarning: Could not infer format, so each element will be parsed individually, falling back to `dateutil`. To ensure parsing is consistent and as-expected, please specify a format.\n",
      "  df['eventDate'] = pd.to_datetime(df['eventDate'])\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Kingdom</th>\n",
       "      <th>Phylum</th>\n",
       "      <th>Class</th>\n",
       "      <th>Order</th>\n",
       "      <th>Family</th>\n",
       "      <th>Genus</th>\n",
       "      <th>Species</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ASV</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>ASV_1</th>\n",
       "      <td>Eukaryota</td>\n",
       "      <td>Chordata</td>\n",
       "      <td>Actinopteri</td>\n",
       "      <td>Clupeiformes</td>\n",
       "      <td>Engraulidae</td>\n",
       "      <td>Engraulis</td>\n",
       "      <td>Engraulis mordax</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ASV_2</th>\n",
       "      <td>Eukaryota</td>\n",
       "      <td>Chordata</td>\n",
       "      <td>Actinopteri</td>\n",
       "      <td>Myctophiformes</td>\n",
       "      <td>Myctophidae</td>\n",
       "      <td>Diaphus</td>\n",
       "      <td>Diaphus theta</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ASV_3</th>\n",
       "      <td>Eukaryota</td>\n",
       "      <td>Chordata</td>\n",
       "      <td>Actinopteri</td>\n",
       "      <td>Gadiformes</td>\n",
       "      <td>Macrouridae</td>\n",
       "      <td>unassigned</td>\n",
       "      <td>unassigned</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ASV_4</th>\n",
       "      <td>Eukaryota</td>\n",
       "      <td>Chordata</td>\n",
       "      <td>Actinopteri</td>\n",
       "      <td>Gadiformes</td>\n",
       "      <td>Merlucciidae</td>\n",
       "      <td>Merluccius</td>\n",
       "      <td>Merluccius productus</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ASV_5</th>\n",
       "      <td>Eukaryota</td>\n",
       "      <td>Chordata</td>\n",
       "      <td>Actinopteri</td>\n",
       "      <td>Myctophiformes</td>\n",
       "      <td>Myctophidae</td>\n",
       "      <td>Stenobrachius</td>\n",
       "      <td>Stenobrachius leucopsarus</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "         Kingdom    Phylum        Class           Order        Family  \\\n",
       "ASV                                                                     \n",
       "ASV_1  Eukaryota  Chordata  Actinopteri    Clupeiformes   Engraulidae   \n",
       "ASV_2  Eukaryota  Chordata  Actinopteri  Myctophiformes   Myctophidae   \n",
       "ASV_3  Eukaryota  Chordata  Actinopteri      Gadiformes   Macrouridae   \n",
       "ASV_4  Eukaryota  Chordata  Actinopteri      Gadiformes  Merlucciidae   \n",
       "ASV_5  Eukaryota  Chordata  Actinopteri  Myctophiformes   Myctophidae   \n",
       "\n",
       "               Genus                    Species  \n",
       "ASV                                              \n",
       "ASV_1      Engraulis           Engraulis mordax  \n",
       "ASV_2        Diaphus              Diaphus theta  \n",
       "ASV_3     unassigned                 unassigned  \n",
       "ASV_4     Merluccius       Merluccius productus  \n",
       "ASV_5  Stenobrachius  Stenobrachius leucopsarus  "
      ]
     },
     "execution_count": 26,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#location of data:\n",
    "directory = '../data/Dada2_seq_data/'\n",
    "markers = ['12S']\n",
    "prefix = 'CN19S'\n",
    "otus= []\n",
    "taxas=[]\n",
    "metas=[]\n",
    "for marker in markers:\n",
    "    print('XXXXXX    ',marker)\n",
    "    df = pd.read_csv(directory+prefix+'_'+marker+'_Dada2_otu_merged.csv')\n",
    "    df = df.rename(columns={'Unnamed: 0':'ASV'})\n",
    "    df.set_index('ASV', inplace=True)\n",
    "    otus.append(df)\n",
    "    print('Number samples in otu_table:', len(list(df)))\n",
    "    df = pd.read_csv(directory+prefix+'_'+marker+'_Dada2_meta_merged.csv')\n",
    "    df.set_index('sample_name', inplace=True)\n",
    "    #date handling\n",
    "    print(list(df))\n",
    "    df['eventDate'] = pd.to_datetime(df['eventDate'])\n",
    "    df['month'] = df['eventDate'].dt.month\n",
    "    metas.append(df)\n",
    "    print('Number samples in metadata table:', len(df.index))\n",
    "    df = pd.read_csv(directory+prefix+'_'+marker+'_Dada2_taxa_merged.csv')\n",
    "    df = df.rename(columns={'Unnamed: 0':'ASV'})\n",
    "    df.set_index('ASV', inplace=True)\n",
    "    print('Number ASVs in taxa table:', len(df.index))\n",
    "    taxas.append(df)\n",
    "taxas[0].head()\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "id": "4f792b12-9eea-4127-9383-3ac75e2db745",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>FilterID</th>\n",
       "      <th>target_gene</th>\n",
       "      <th>PlateID</th>\n",
       "      <th>library</th>\n",
       "      <th>local_time</th>\n",
       "      <th>time_label</th>\n",
       "      <th>SAMPLING_cruise</th>\n",
       "      <th>depth</th>\n",
       "      <th>SAMPLING_platform</th>\n",
       "      <th>SC</th>\n",
       "      <th>...</th>\n",
       "      <th>chlorophyll</th>\n",
       "      <th>bbp470 (count)</th>\n",
       "      <th>bbp650 (count)</th>\n",
       "      <th>SAMPLING_project</th>\n",
       "      <th>ESP_name</th>\n",
       "      <th>diel</th>\n",
       "      <th>month</th>\n",
       "      <th>day</th>\n",
       "      <th>hour</th>\n",
       "      <th>eventDate</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>sample_name</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>CN19SESPMV1_SC58_eDNA_CE</th>\n",
       "      <td>CN19SESPMV1_SC58_eDNA</td>\n",
       "      <td>12S</td>\n",
       "      <td>CE</td>\n",
       "      <td>CE40</td>\n",
       "      <td>2019-05-29 02:42:00</td>\n",
       "      <td>05-29 night</td>\n",
       "      <td>CN19S</td>\n",
       "      <td>17.6279</td>\n",
       "      <td>daphne</td>\n",
       "      <td>58.0</td>\n",
       "      <td>...</td>\n",
       "      <td>2.7972</td>\n",
       "      <td>100.9675</td>\n",
       "      <td>109.2667</td>\n",
       "      <td>CANON</td>\n",
       "      <td>NaN</td>\n",
       "      <td>night</td>\n",
       "      <td>5.0</td>\n",
       "      <td>29.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2019-05-29 09:42:00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CN19SESPKOA_SC58_eDNA_CE</th>\n",
       "      <td>CN19SESPKOA_SC58_eDNA</td>\n",
       "      <td>12S</td>\n",
       "      <td>CE</td>\n",
       "      <td>CE3</td>\n",
       "      <td>2019-05-29 12:32:00</td>\n",
       "      <td>05-29 day</td>\n",
       "      <td>CN19S</td>\n",
       "      <td>32.5799</td>\n",
       "      <td>makai</td>\n",
       "      <td>58.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.5502</td>\n",
       "      <td>83.3809</td>\n",
       "      <td>149.0871</td>\n",
       "      <td>CANON</td>\n",
       "      <td>NaN</td>\n",
       "      <td>day</td>\n",
       "      <td>5.0</td>\n",
       "      <td>29.0</td>\n",
       "      <td>12.0</td>\n",
       "      <td>2019-05-29 19:32:00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CN19SESPKOA_SC57_eDNA_CE</th>\n",
       "      <td>CN19SESPKOA_SC57_eDNA</td>\n",
       "      <td>12S</td>\n",
       "      <td>CE</td>\n",
       "      <td>CE4</td>\n",
       "      <td>2019-05-29 13:47:00</td>\n",
       "      <td>05-29 day</td>\n",
       "      <td>CN19S</td>\n",
       "      <td>202.5248</td>\n",
       "      <td>makai</td>\n",
       "      <td>57.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0782</td>\n",
       "      <td>78.4872</td>\n",
       "      <td>146.0477</td>\n",
       "      <td>CANON</td>\n",
       "      <td>NaN</td>\n",
       "      <td>day</td>\n",
       "      <td>5.0</td>\n",
       "      <td>29.0</td>\n",
       "      <td>13.0</td>\n",
       "      <td>2019-05-29 20:47:00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CN19SESPKOA_SC56_eDNA_CE</th>\n",
       "      <td>CN19SESPKOA_SC56_eDNA</td>\n",
       "      <td>12S</td>\n",
       "      <td>CE</td>\n",
       "      <td>CE5</td>\n",
       "      <td>2019-05-29 14:50:00</td>\n",
       "      <td>05-29 day</td>\n",
       "      <td>CN19S</td>\n",
       "      <td>267.6256</td>\n",
       "      <td>makai</td>\n",
       "      <td>56.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0833</td>\n",
       "      <td>78.2715</td>\n",
       "      <td>148.8699</td>\n",
       "      <td>CANON</td>\n",
       "      <td>NaN</td>\n",
       "      <td>day</td>\n",
       "      <td>5.0</td>\n",
       "      <td>29.0</td>\n",
       "      <td>14.0</td>\n",
       "      <td>2019-05-29 21:50:00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CN19SESPKOA_SC55_eDNA_CE</th>\n",
       "      <td>CN19SESPKOA_SC55_eDNA</td>\n",
       "      <td>12S</td>\n",
       "      <td>CE</td>\n",
       "      <td>CE6</td>\n",
       "      <td>2019-05-29 23:17:00</td>\n",
       "      <td>05-29 night</td>\n",
       "      <td>CN19S</td>\n",
       "      <td>27.4770</td>\n",
       "      <td>makai</td>\n",
       "      <td>55.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.3254</td>\n",
       "      <td>80.2178</td>\n",
       "      <td>143.2971</td>\n",
       "      <td>CANON</td>\n",
       "      <td>NaN</td>\n",
       "      <td>night</td>\n",
       "      <td>5.0</td>\n",
       "      <td>29.0</td>\n",
       "      <td>23.0</td>\n",
       "      <td>2019-05-30 06:17:00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ArtComm1_CE</th>\n",
       "      <td>ArtComm1</td>\n",
       "      <td>12S</td>\n",
       "      <td>CE</td>\n",
       "      <td>CE92</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ArtComm2_CE</th>\n",
       "      <td>ArtComm2</td>\n",
       "      <td>12S</td>\n",
       "      <td>CE</td>\n",
       "      <td>CE93</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CN19SP3_OSU_EB1_CE</th>\n",
       "      <td>CN19SP3_OSU_EB1</td>\n",
       "      <td>12S</td>\n",
       "      <td>CE</td>\n",
       "      <td>CE94</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CN19SP3_OSU_EB2_CE</th>\n",
       "      <td>CN19SP3_OSU_EB2</td>\n",
       "      <td>12S</td>\n",
       "      <td>CE</td>\n",
       "      <td>CE95</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>RTSF_NTC_1284_CE</th>\n",
       "      <td>RTSF_NTC_1284</td>\n",
       "      <td>12S</td>\n",
       "      <td>CE</td>\n",
       "      <td>CE96</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaT</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>286 rows × 50 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                       FilterID target_gene PlateID library  \\\n",
       "sample_name                                                                   \n",
       "CN19SESPMV1_SC58_eDNA_CE  CN19SESPMV1_SC58_eDNA         12S      CE    CE40   \n",
       "CN19SESPKOA_SC58_eDNA_CE  CN19SESPKOA_SC58_eDNA         12S      CE     CE3   \n",
       "CN19SESPKOA_SC57_eDNA_CE  CN19SESPKOA_SC57_eDNA         12S      CE     CE4   \n",
       "CN19SESPKOA_SC56_eDNA_CE  CN19SESPKOA_SC56_eDNA         12S      CE     CE5   \n",
       "CN19SESPKOA_SC55_eDNA_CE  CN19SESPKOA_SC55_eDNA         12S      CE     CE6   \n",
       "...                                         ...         ...     ...     ...   \n",
       "ArtComm1_CE                            ArtComm1         12S      CE    CE92   \n",
       "ArtComm2_CE                            ArtComm2         12S      CE    CE93   \n",
       "CN19SP3_OSU_EB1_CE              CN19SP3_OSU_EB1         12S      CE    CE94   \n",
       "CN19SP3_OSU_EB2_CE              CN19SP3_OSU_EB2         12S      CE    CE95   \n",
       "RTSF_NTC_1284_CE                  RTSF_NTC_1284         12S      CE    CE96   \n",
       "\n",
       "                                   local_time   time_label SAMPLING_cruise  \\\n",
       "sample_name                                                                  \n",
       "CN19SESPMV1_SC58_eDNA_CE  2019-05-29 02:42:00  05-29 night           CN19S   \n",
       "CN19SESPKOA_SC58_eDNA_CE  2019-05-29 12:32:00    05-29 day           CN19S   \n",
       "CN19SESPKOA_SC57_eDNA_CE  2019-05-29 13:47:00    05-29 day           CN19S   \n",
       "CN19SESPKOA_SC56_eDNA_CE  2019-05-29 14:50:00    05-29 day           CN19S   \n",
       "CN19SESPKOA_SC55_eDNA_CE  2019-05-29 23:17:00  05-29 night           CN19S   \n",
       "...                                       ...          ...             ...   \n",
       "ArtComm1_CE                               NaN          NaN             NaN   \n",
       "ArtComm2_CE                               NaN          NaN             NaN   \n",
       "CN19SP3_OSU_EB1_CE                        NaN          NaN             NaN   \n",
       "CN19SP3_OSU_EB2_CE                        NaN          NaN             NaN   \n",
       "RTSF_NTC_1284_CE                          NaN          NaN             NaN   \n",
       "\n",
       "                             depth SAMPLING_platform    SC  ... chlorophyll  \\\n",
       "sample_name                                                 ...               \n",
       "CN19SESPMV1_SC58_eDNA_CE   17.6279            daphne  58.0  ...      2.7972   \n",
       "CN19SESPKOA_SC58_eDNA_CE   32.5799             makai  58.0  ...      0.5502   \n",
       "CN19SESPKOA_SC57_eDNA_CE  202.5248             makai  57.0  ...      0.0782   \n",
       "CN19SESPKOA_SC56_eDNA_CE  267.6256             makai  56.0  ...      0.0833   \n",
       "CN19SESPKOA_SC55_eDNA_CE   27.4770             makai  55.0  ...      0.3254   \n",
       "...                            ...               ...   ...  ...         ...   \n",
       "ArtComm1_CE                    NaN               NaN   NaN  ...         NaN   \n",
       "ArtComm2_CE                    NaN               NaN   NaN  ...         NaN   \n",
       "CN19SP3_OSU_EB1_CE             NaN               NaN   NaN  ...         NaN   \n",
       "CN19SP3_OSU_EB2_CE             NaN               NaN   NaN  ...         NaN   \n",
       "RTSF_NTC_1284_CE               NaN               NaN   NaN  ...         NaN   \n",
       "\n",
       "                          bbp470 (count) bbp650 (count)  SAMPLING_project  \\\n",
       "sample_name                                                                 \n",
       "CN19SESPMV1_SC58_eDNA_CE        100.9675       109.2667             CANON   \n",
       "CN19SESPKOA_SC58_eDNA_CE         83.3809       149.0871             CANON   \n",
       "CN19SESPKOA_SC57_eDNA_CE         78.4872       146.0477             CANON   \n",
       "CN19SESPKOA_SC56_eDNA_CE         78.2715       148.8699             CANON   \n",
       "CN19SESPKOA_SC55_eDNA_CE         80.2178       143.2971             CANON   \n",
       "...                                  ...            ...               ...   \n",
       "ArtComm1_CE                          NaN            NaN               NaN   \n",
       "ArtComm2_CE                          NaN            NaN               NaN   \n",
       "CN19SP3_OSU_EB1_CE                   NaN            NaN               NaN   \n",
       "CN19SP3_OSU_EB2_CE                   NaN            NaN               NaN   \n",
       "RTSF_NTC_1284_CE                     NaN            NaN               NaN   \n",
       "\n",
       "                          ESP_name   diel month   day  hour  \\\n",
       "sample_name                                                   \n",
       "CN19SESPMV1_SC58_eDNA_CE       NaN  night   5.0  29.0   2.0   \n",
       "CN19SESPKOA_SC58_eDNA_CE       NaN    day   5.0  29.0  12.0   \n",
       "CN19SESPKOA_SC57_eDNA_CE       NaN    day   5.0  29.0  13.0   \n",
       "CN19SESPKOA_SC56_eDNA_CE       NaN    day   5.0  29.0  14.0   \n",
       "CN19SESPKOA_SC55_eDNA_CE       NaN  night   5.0  29.0  23.0   \n",
       "...                            ...    ...   ...   ...   ...   \n",
       "ArtComm1_CE                    NaN    NaN   NaN   NaN   NaN   \n",
       "ArtComm2_CE                    NaN    NaN   NaN   NaN   NaN   \n",
       "CN19SP3_OSU_EB1_CE             NaN    NaN   NaN   NaN   NaN   \n",
       "CN19SP3_OSU_EB2_CE             NaN    NaN   NaN   NaN   NaN   \n",
       "RTSF_NTC_1284_CE               NaN    NaN   NaN   NaN   NaN   \n",
       "\n",
       "                                   eventDate  \n",
       "sample_name                                   \n",
       "CN19SESPMV1_SC58_eDNA_CE 2019-05-29 09:42:00  \n",
       "CN19SESPKOA_SC58_eDNA_CE 2019-05-29 19:32:00  \n",
       "CN19SESPKOA_SC57_eDNA_CE 2019-05-29 20:47:00  \n",
       "CN19SESPKOA_SC56_eDNA_CE 2019-05-29 21:50:00  \n",
       "CN19SESPKOA_SC55_eDNA_CE 2019-05-30 06:17:00  \n",
       "...                                      ...  \n",
       "ArtComm1_CE                              NaT  \n",
       "ArtComm2_CE                              NaT  \n",
       "CN19SP3_OSU_EB1_CE                       NaT  \n",
       "CN19SP3_OSU_EB2_CE                       NaT  \n",
       "RTSF_NTC_1284_CE                         NaT  \n",
       "\n",
       "[286 rows x 50 columns]"
      ]
     },
     "execution_count": 27,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "metas[0]"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "c02765c6-318a-46cd-91b3-523df3a2bdf6",
   "metadata": {},
   "source": [
    "## Get Filenames data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 52,
   "id": "f6341509-bf81-4b03-b03b-aed883e41b50",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "['ESP', 'PCR_settings', 'PlateID', 'R1', 'R2', 'SAMPLING_PI', 'SAMPLING_bottle', 'SAMPLING_campaign', 'SAMPLING_cruise', 'SAMPLING_date_time', 'SAMPLING_dec_lat', 'SAMPLING_dec_long', 'SAMPLING_institute', 'SAMPLING_platform', 'SAMPLING_platform_type', 'SAMPLING_project', 'SAMPLING_rdepth', 'SAMPLING_real_depth', 'SAMPLING_station', 'SAMPLING_station_number', 'SC', 'Unnamed: 0', 'chlorophyll', 'collection_date', 'date_pcr', 'day', 'decimalLatitude', 'decimalLongitude', 'density', 'depth', 'diss_oxygen', 'end_GMT', 'env_biome', 'env_broad_scale', 'env_feature', 'env_local_scale', 'env_material', 'env_medium', 'env_package', 'eventDate', 'fluor', 'geo_loc_name', 'investigation_type', 'library', 'library_tag_combo', 'maximumDepthInMeters', 'minimumDepthInMeters', 'month', 'nitrate', 'order', 'original_name', 'pcr_primer_name_forward', 'pcr_primer_name_reverse', 'pcr_primer_reference', 'pressure', 'pressure_dbar', 'primer_sequence_F', 'primer_sequence_R', 'primer_sequence_forward', 'primer_sequence_reverse', 'project_name', 'salinity', 'samp_filter_ext_type', 'samp_filter_size_ext', 'samp_store_temp', 'sample_locus', 'sample_type', 'seqID', 'seq_meth', 'sequencing_facility', 'start_GMT', 'tag_number', 'tag_sequence', 'target_gene', 'tdepth', 'temp', 'year', 'pcr_label']\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>primer_sequence_forward</th>\n",
       "      <th>primer_sequence_reverse</th>\n",
       "      <th>diss_oxygen</th>\n",
       "      <th>samp_store_temp</th>\n",
       "      <th>samp_filter_ext_type</th>\n",
       "      <th>samp_filter_size_ext</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>sample_name</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>V4195_CB_eDNA_BT</th>\n",
       "      <td>GTCGGTAAAACTCGTGCCAGC</td>\n",
       "      <td>CATAGTGGGGTATCTAATCCCAGTTTG</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-80</td>\n",
       "      <td>Poretics</td>\n",
       "      <td>0.22um</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CN19Sc24_04_eDNA_BT</th>\n",
       "      <td>GTCGGTAAAACTCGTGCCAGC</td>\n",
       "      <td>CATAGTGGGGTATCTAATCCCAGTTTG</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-80</td>\n",
       "      <td>Poretics</td>\n",
       "      <td>0.22um</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CN19Sc24_09_eDNA_BT</th>\n",
       "      <td>GTCGGTAAAACTCGTGCCAGC</td>\n",
       "      <td>CATAGTGGGGTATCTAATCCCAGTTTG</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-80</td>\n",
       "      <td>Poretics</td>\n",
       "      <td>0.22um</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CN19Sc26_02_eDNA_BT</th>\n",
       "      <td>GTCGGTAAAACTCGTGCCAGC</td>\n",
       "      <td>CATAGTGGGGTATCTAATCCCAGTTTG</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-80</td>\n",
       "      <td>Poretics</td>\n",
       "      <td>0.22um</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CN19Sc26_04_eDNA_BT</th>\n",
       "      <td>GTCGGTAAAACTCGTGCCAGC</td>\n",
       "      <td>CATAGTGGGGTATCTAATCCCAGTTTG</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-80</td>\n",
       "      <td>Poretics</td>\n",
       "      <td>0.22um</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ArtComm1_CE</th>\n",
       "      <td>GTCGGTAAAACTCGTGCCAGC</td>\n",
       "      <td>CATAGTGGGGTATCTAATCCCAGTTTG</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-80</td>\n",
       "      <td>Poretics</td>\n",
       "      <td>0.22um</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ArtComm2_CE</th>\n",
       "      <td>GTCGGTAAAACTCGTGCCAGC</td>\n",
       "      <td>CATAGTGGGGTATCTAATCCCAGTTTG</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-80</td>\n",
       "      <td>Poretics</td>\n",
       "      <td>0.22um</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CN19SP3_OSU_EB1_CE</th>\n",
       "      <td>GTCGGTAAAACTCGTGCCAGC</td>\n",
       "      <td>CATAGTGGGGTATCTAATCCCAGTTTG</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-80</td>\n",
       "      <td>Poretics</td>\n",
       "      <td>0.22um</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CN19SP3_OSU_EB2_CE</th>\n",
       "      <td>GTCGGTAAAACTCGTGCCAGC</td>\n",
       "      <td>CATAGTGGGGTATCTAATCCCAGTTTG</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-80</td>\n",
       "      <td>Poretics</td>\n",
       "      <td>0.22um</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>RTSF_NTC_1284_CE</th>\n",
       "      <td>GTCGGTAAAACTCGTGCCAGC</td>\n",
       "      <td>CATAGTGGGGTATCTAATCCCAGTTTG</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-80</td>\n",
       "      <td>Poretics</td>\n",
       "      <td>0.22um</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>286 rows × 6 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                    primer_sequence_forward      primer_sequence_reverse  \\\n",
       "sample_name                                                                \n",
       "V4195_CB_eDNA_BT      GTCGGTAAAACTCGTGCCAGC  CATAGTGGGGTATCTAATCCCAGTTTG   \n",
       "CN19Sc24_04_eDNA_BT   GTCGGTAAAACTCGTGCCAGC  CATAGTGGGGTATCTAATCCCAGTTTG   \n",
       "CN19Sc24_09_eDNA_BT   GTCGGTAAAACTCGTGCCAGC  CATAGTGGGGTATCTAATCCCAGTTTG   \n",
       "CN19Sc26_02_eDNA_BT   GTCGGTAAAACTCGTGCCAGC  CATAGTGGGGTATCTAATCCCAGTTTG   \n",
       "CN19Sc26_04_eDNA_BT   GTCGGTAAAACTCGTGCCAGC  CATAGTGGGGTATCTAATCCCAGTTTG   \n",
       "...                                     ...                          ...   \n",
       "ArtComm1_CE           GTCGGTAAAACTCGTGCCAGC  CATAGTGGGGTATCTAATCCCAGTTTG   \n",
       "ArtComm2_CE           GTCGGTAAAACTCGTGCCAGC  CATAGTGGGGTATCTAATCCCAGTTTG   \n",
       "CN19SP3_OSU_EB1_CE    GTCGGTAAAACTCGTGCCAGC  CATAGTGGGGTATCTAATCCCAGTTTG   \n",
       "CN19SP3_OSU_EB2_CE    GTCGGTAAAACTCGTGCCAGC  CATAGTGGGGTATCTAATCCCAGTTTG   \n",
       "RTSF_NTC_1284_CE      GTCGGTAAAACTCGTGCCAGC  CATAGTGGGGTATCTAATCCCAGTTTG   \n",
       "\n",
       "                     diss_oxygen samp_store_temp samp_filter_ext_type  \\\n",
       "sample_name                                                             \n",
       "V4195_CB_eDNA_BT             NaN             -80             Poretics   \n",
       "CN19Sc24_04_eDNA_BT          NaN             -80             Poretics   \n",
       "CN19Sc24_09_eDNA_BT          NaN             -80             Poretics   \n",
       "CN19Sc26_02_eDNA_BT          NaN             -80             Poretics   \n",
       "CN19Sc26_04_eDNA_BT          NaN             -80             Poretics   \n",
       "...                          ...             ...                  ...   \n",
       "ArtComm1_CE                  NaN             -80             Poretics   \n",
       "ArtComm2_CE                  NaN             -80             Poretics   \n",
       "CN19SP3_OSU_EB1_CE           NaN             -80             Poretics   \n",
       "CN19SP3_OSU_EB2_CE           NaN             -80             Poretics   \n",
       "RTSF_NTC_1284_CE             NaN             -80             Poretics   \n",
       "\n",
       "                    samp_filter_size_ext  \n",
       "sample_name                               \n",
       "V4195_CB_eDNA_BT                  0.22um  \n",
       "CN19Sc24_04_eDNA_BT               0.22um  \n",
       "CN19Sc24_09_eDNA_BT               0.22um  \n",
       "CN19Sc26_02_eDNA_BT               0.22um  \n",
       "CN19Sc26_04_eDNA_BT               0.22um  \n",
       "...                                  ...  \n",
       "ArtComm1_CE                       0.22um  \n",
       "ArtComm2_CE                       0.22um  \n",
       "CN19SP3_OSU_EB1_CE                0.22um  \n",
       "CN19SP3_OSU_EB2_CE                0.22um  \n",
       "RTSF_NTC_1284_CE                  0.22um  \n",
       "\n",
       "[286 rows x 6 columns]"
      ]
     },
     "execution_count": 52,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "marker = '12S'\n",
    "merged_directory = '/Volumes/MBON/processed/banzai_Dada2/12S/Merged_dataset/Results_20220824/'\n",
    "file = 'Collapsed_meta_table_unfiltered.csv'\n",
    "# import and filter to samples in this study\n",
    "df = pd.read_csv(merged_directory + file)\n",
    "# samples in study:\n",
    "study_samps = metas[0].index.tolist()\n",
    "df = df.loc[df['sample_name'].isin(study_samps)==True]\n",
    "df.set_index('sample_name', inplace=True)\n",
    "df = df.dropna(how='all', axis=1)\n",
    "print(list(df))\n",
    "meta_all = df.copy()\n",
    "meta_all[['primer_sequence_forward', 'primer_sequence_reverse', 'diss_oxygen', 'samp_store_temp', 'samp_filter_ext_type', 'samp_filter_size_ext']]\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "58d086a6-660a-48a9-9057-4b53581d751b",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "markdown",
   "id": "832d6b7d-0ff9-4458-9ce5-81fdbe3feecc",
   "metadata": {},
   "source": [
    "## Import Filtration metadata"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "id": "e3fc9181-4c3e-4baf-89af-939bb4b37f29",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>extraction_date</th>\n",
       "      <th>water_filt_mL</th>\n",
       "      <th>DNA_concentration_ng_uL</th>\n",
       "      <th>measurement_method</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>sample_name</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>CN19Sc03_1_eDNA_JJ</th>\n",
       "      <td>20190916</td>\n",
       "      <td>900.000</td>\n",
       "      <td>1.44</td>\n",
       "      <td>qubit</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CN19Sc03_10_eDNA_JJ</th>\n",
       "      <td>20190916</td>\n",
       "      <td>1000.000</td>\n",
       "      <td>16.10</td>\n",
       "      <td>nanodrop</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CN19Sc03_12_eDNA_JJ</th>\n",
       "      <td>20190916</td>\n",
       "      <td>1000.000</td>\n",
       "      <td>68.00</td>\n",
       "      <td>nanodrop</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CN19Sc03_3_eDNA_JJ</th>\n",
       "      <td>20190916</td>\n",
       "      <td>1000.000</td>\n",
       "      <td>4.20</td>\n",
       "      <td>nanodrop</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CN19Sc03_4_eDNA_JJ</th>\n",
       "      <td>20190916</td>\n",
       "      <td>700.000</td>\n",
       "      <td>5.10</td>\n",
       "      <td>nanodrop</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CN19SESPMV1_SC52_eDNA_CE</th>\n",
       "      <td>20200814</td>\n",
       "      <td>1000.000</td>\n",
       "      <td>7.86</td>\n",
       "      <td>qubit</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CN19SESPMV1_SC53_eDNA_CE</th>\n",
       "      <td>20200814</td>\n",
       "      <td>484.645</td>\n",
       "      <td>3.76</td>\n",
       "      <td>qubit</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CN19SESPMV1_SC54_eDNA_CE</th>\n",
       "      <td>20200814</td>\n",
       "      <td>1000.000</td>\n",
       "      <td>3.99</td>\n",
       "      <td>qubit</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CN19SESPMV1_SC55_eDNA_CE</th>\n",
       "      <td>20200814</td>\n",
       "      <td>1000.000</td>\n",
       "      <td>4.86</td>\n",
       "      <td>qubit</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CN19SESPMV1_SC58_eDNA_CE</th>\n",
       "      <td>20200814</td>\n",
       "      <td>486.135</td>\n",
       "      <td>14.40</td>\n",
       "      <td>qubit</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>245 rows × 4 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                          extraction_date  water_filt_mL  \\\n",
       "sample_name                                                \n",
       "CN19Sc03_1_eDNA_JJ               20190916        900.000   \n",
       "CN19Sc03_10_eDNA_JJ              20190916       1000.000   \n",
       "CN19Sc03_12_eDNA_JJ              20190916       1000.000   \n",
       "CN19Sc03_3_eDNA_JJ               20190916       1000.000   \n",
       "CN19Sc03_4_eDNA_JJ               20190916        700.000   \n",
       "...                                   ...            ...   \n",
       "CN19SESPMV1_SC52_eDNA_CE         20200814       1000.000   \n",
       "CN19SESPMV1_SC53_eDNA_CE         20200814        484.645   \n",
       "CN19SESPMV1_SC54_eDNA_CE         20200814       1000.000   \n",
       "CN19SESPMV1_SC55_eDNA_CE         20200814       1000.000   \n",
       "CN19SESPMV1_SC58_eDNA_CE         20200814        486.135   \n",
       "\n",
       "                          DNA_concentration_ng_uL measurement_method  \n",
       "sample_name                                                           \n",
       "CN19Sc03_1_eDNA_JJ                           1.44              qubit  \n",
       "CN19Sc03_10_eDNA_JJ                         16.10           nanodrop  \n",
       "CN19Sc03_12_eDNA_JJ                         68.00           nanodrop  \n",
       "CN19Sc03_3_eDNA_JJ                           4.20           nanodrop  \n",
       "CN19Sc03_4_eDNA_JJ                           5.10           nanodrop  \n",
       "...                                           ...                ...  \n",
       "CN19SESPMV1_SC52_eDNA_CE                     7.86              qubit  \n",
       "CN19SESPMV1_SC53_eDNA_CE                     3.76              qubit  \n",
       "CN19SESPMV1_SC54_eDNA_CE                     3.99              qubit  \n",
       "CN19SESPMV1_SC55_eDNA_CE                     4.86              qubit  \n",
       "CN19SESPMV1_SC58_eDNA_CE                    14.40              qubit  \n",
       "\n",
       "[245 rows x 4 columns]"
      ]
     },
     "execution_count": 28,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# environmental samples\n",
    "file = '../data/metadata/CN19S_extraction_metadata.csv'\n",
    "df = pd.read_csv(file)\n",
    "df.set_index('sample_name', inplace=True)\n",
    "extract = df.copy()\n",
    "df"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "fa84680f-0e25-47e8-8252-cd179d3c3bcb",
   "metadata": {},
   "source": [
    "## Join and make metadata columns needed by SRA"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 95,
   "id": "f79099b9-e252-4c16-a45c-6ac2e671ec16",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "['seawater metagenome' 'control sample' 'synthetic metagenome']\n",
      "['Environmental Sample Processor onboard long-range autonomous underwater vehicle (LRAUV-ESP)'\n",
      " 'CTD rosette' nan]\n",
      "['Seawater was filtered onto 0.22 PVDF membrane filters through shipboard and vehicle sampling. DNA was extracted using the Qiagen DNeasy Blood and Tissue Kit. A two-step PCR approach was used, targeting mitochondrial 12S with primers MiFish-U-F (5-GTCGGTAAAACTCGTGCCAGC-3) and MiFish-U-R (5-CATAGTGGGGTATCTAATCCCAGTTTG-3). Fluidigm tail sequences CS1 (5-ACACTGACGACATGGTTCTACA-3) and CS2 (5-TACGGTAGCAGAGACTTGGTCT-3) were added to the 5-prime ends of the forward and reverse primer sequences. Secondary amplification and next-generation sequencing were performed at Michigan State University where Illumina compatible adapters with barcodes targeted CS1/CS2 ends of the primery PCR products. Pooled product was size selected with a Pippin HT to select for the vertebrate/fish band (~350bp) and remove co-amplified bacteria (~435bp). It was then sequenced in a 2x250bp paired end format on an Illumina MiSeq.'\n",
      " 'Seawater was filtered onto 0.22 PVDF membrane filters through shipboard and vehicle sampling. DNA was extracted using the Qiagen DNeasy Blood and Tissue Kit. A two-step PCR approach was used, targeting mitochondrial 12S with primers MiFish-U-F (5-GTCGGTAAAACTCGTGCCAGC-3) and MiFish-U-R (5-CATAGTGGGGTATCTAATCCCAGTTTG-3). Fluidigm tail sequences CS1 (5-ACACTGACGACATGGTTCTACA-3) and CS2 (5-TACGGTAGCAGAGACTTGGTCT-3) were added to the 5-prime ends of the forward and reverse primer sequences. Secondary amplification and next-generation sequencing were performed at Michigan State University where Illumina compatible adapters with barcodes targeted CS1/CS2 ends of the primery PCR products. Pooled product was sequenced in a 2x250bp paired end format on an Illumina MiSeq.']\n",
      "density\n",
      "kilogram per cubic meter\n",
      "chlorophyll\n",
      "milligram per cubic meter\n",
      "depth\n",
      "meter\n",
      "diss_oxygen\n",
      "milliliter per liter\n",
      "fluor\n",
      "volts\n",
      "nitrate\n",
      "micromole per liter\n",
      "pressure\n",
      "decibar\n",
      "salinity\n",
      "parts per thousand\n",
      "samp_vol_we_dna_ext\n",
      "milliliter\n",
      "temp\n",
      "degree Celsius\n",
      "dna_conc\n",
      "\n",
      "255\n",
      "245\n",
      "['CE' 'JJ' 'BT' 'RR']\n",
      "Done!\n",
      "['ESP' 'CTD']\n",
      "['target_gene', 'PlateID', 'library_ID', 'Sampling_method', 'replicate', 'SAMPLING_rdepth', 'start_GMT', 'end_GMT', 'sigmat', 'spice', 'PAR (umol/s/m2)', 'altitude', 'bbp470 (count)', 'bbp650 (count)', 'SAMPLING_project', 'eventDate', 'extraction_date', 'filename', 'filename2', 'size_frac', 'decimalLongitude_str', 'title', 'library_strategy', 'library_source', 'library_selection', 'library_layout', 'platform', 'instrument_model', 'design_description', 'filetype', 'concentrationUnit']\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/var/folders/5v/h4s_0pz17yb6f6krsxkzr5dw000b65/T/ipykernel_84247/3943633351.py:40: FutureWarning: Setting an item of incompatible dtype is deprecated and will raise in a future error of pandas. Value 'marine biome [ENVO:00000447]' has dtype incompatible with float64, please explicitly cast to a compatible dtype first.\n",
      "  df.loc[df['organism']=='seawater metagenome','env_broad_scale' ] = 'marine biome [ENVO:00000447]'\n",
      "/var/folders/5v/h4s_0pz17yb6f6krsxkzr5dw000b65/T/ipykernel_84247/3943633351.py:41: FutureWarning: Setting an item of incompatible dtype is deprecated and will raise in a future error of pandas. Value 'marine mesopelagic zone [ENVO:00000213]' has dtype incompatible with float64, please explicitly cast to a compatible dtype first.\n",
      "  df.loc[df['organism']=='seawater metagenome','env_local_scale' ] = 'marine mesopelagic zone [ENVO:00000213]'\n",
      "/var/folders/5v/h4s_0pz17yb6f6krsxkzr5dw000b65/T/ipykernel_84247/3943633351.py:42: FutureWarning: Setting an item of incompatible dtype is deprecated and will raise in a future error of pandas. Value 'sea water [ENVO:00002149]' has dtype incompatible with float64, please explicitly cast to a compatible dtype first.\n",
      "  df.loc[df['organism']=='seawater metagenome','env_medium' ] = 'sea water [ENVO:00002149]'\n",
      "/var/folders/5v/h4s_0pz17yb6f6krsxkzr5dw000b65/T/ipykernel_84247/3943633351.py:43: FutureWarning: Setting an item of incompatible dtype is deprecated and will raise in a future error of pandas. Value 'USA: Monterey Bay' has dtype incompatible with float64, please explicitly cast to a compatible dtype first.\n",
      "  df.loc[df['organism']=='seawater metagenome','geo_loc_name' ] = 'USA: Monterey Bay'\n",
      "/var/folders/5v/h4s_0pz17yb6f6krsxkzr5dw000b65/T/ipykernel_84247/3943633351.py:44: FutureWarning: Setting an item of incompatible dtype is deprecated and will raise in a future error of pandas. Value '0.22 micrometer' has dtype incompatible with float64, please explicitly cast to a compatible dtype first.\n",
      "  df.loc[df['organism']=='seawater metagenome','size_frac' ] = '0.22 micrometer'\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>target_gene</th>\n",
       "      <th>PlateID</th>\n",
       "      <th>library_ID</th>\n",
       "      <th>Sampling_method</th>\n",
       "      <th>replicate</th>\n",
       "      <th>SAMPLING_rdepth</th>\n",
       "      <th>start_GMT</th>\n",
       "      <th>end_GMT</th>\n",
       "      <th>sigmat</th>\n",
       "      <th>spice</th>\n",
       "      <th>...</th>\n",
       "      <th>title</th>\n",
       "      <th>library_strategy</th>\n",
       "      <th>library_source</th>\n",
       "      <th>library_selection</th>\n",
       "      <th>library_layout</th>\n",
       "      <th>platform</th>\n",
       "      <th>instrument_model</th>\n",
       "      <th>design_description</th>\n",
       "      <th>filetype</th>\n",
       "      <th>concentrationUnit</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>sample_name</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>CN19SESPMV1_SC58_eDNA</th>\n",
       "      <td>12S</td>\n",
       "      <td>CE</td>\n",
       "      <td>CE40</td>\n",
       "      <td>ESP</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>5/29/19 9:42</td>\n",
       "      <td>5/29/19 10:07</td>\n",
       "      <td>25.0274</td>\n",
       "      <td>0.2338</td>\n",
       "      <td>...</td>\n",
       "      <td>12S amplicon metabarcoding of marine metagenom...</td>\n",
       "      <td>AMPLICON</td>\n",
       "      <td>METAGENOMIC</td>\n",
       "      <td>PCR</td>\n",
       "      <td>paired</td>\n",
       "      <td>ILLUMINA</td>\n",
       "      <td>Illumina MiSeq</td>\n",
       "      <td>Seawater was filtered onto 0.22 PVDF membrane ...</td>\n",
       "      <td>fastq</td>\n",
       "      <td>nanograms per milliliter</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CN19SESPKOA_SC58_eDNA</th>\n",
       "      <td>12S</td>\n",
       "      <td>CE</td>\n",
       "      <td>CE3</td>\n",
       "      <td>ESP</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>5/29/19 19:32</td>\n",
       "      <td>5/29/19 20:18</td>\n",
       "      <td>25.4315</td>\n",
       "      <td>0.0592</td>\n",
       "      <td>...</td>\n",
       "      <td>12S amplicon metabarcoding of marine metagenom...</td>\n",
       "      <td>AMPLICON</td>\n",
       "      <td>METAGENOMIC</td>\n",
       "      <td>PCR</td>\n",
       "      <td>paired</td>\n",
       "      <td>ILLUMINA</td>\n",
       "      <td>Illumina MiSeq</td>\n",
       "      <td>Seawater was filtered onto 0.22 PVDF membrane ...</td>\n",
       "      <td>fastq</td>\n",
       "      <td>nanograms per milliliter</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CN19SESPKOA_SC57_eDNA</th>\n",
       "      <td>12S</td>\n",
       "      <td>CE</td>\n",
       "      <td>CE4</td>\n",
       "      <td>ESP</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>5/29/19 20:47</td>\n",
       "      <td>5/29/19 21:28</td>\n",
       "      <td>26.5216</td>\n",
       "      <td>0.0742</td>\n",
       "      <td>...</td>\n",
       "      <td>12S amplicon metabarcoding of marine metagenom...</td>\n",
       "      <td>AMPLICON</td>\n",
       "      <td>METAGENOMIC</td>\n",
       "      <td>PCR</td>\n",
       "      <td>paired</td>\n",
       "      <td>ILLUMINA</td>\n",
       "      <td>Illumina MiSeq</td>\n",
       "      <td>Seawater was filtered onto 0.22 PVDF membrane ...</td>\n",
       "      <td>fastq</td>\n",
       "      <td>nanograms per milliliter</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CN19SESPKOA_SC56_eDNA</th>\n",
       "      <td>12S</td>\n",
       "      <td>CE</td>\n",
       "      <td>CE5</td>\n",
       "      <td>ESP</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>5/29/19 21:50</td>\n",
       "      <td>5/29/19 22:33</td>\n",
       "      <td>26.6542</td>\n",
       "      <td>0.0243</td>\n",
       "      <td>...</td>\n",
       "      <td>12S amplicon metabarcoding of marine metagenom...</td>\n",
       "      <td>AMPLICON</td>\n",
       "      <td>METAGENOMIC</td>\n",
       "      <td>PCR</td>\n",
       "      <td>paired</td>\n",
       "      <td>ILLUMINA</td>\n",
       "      <td>Illumina MiSeq</td>\n",
       "      <td>Seawater was filtered onto 0.22 PVDF membrane ...</td>\n",
       "      <td>fastq</td>\n",
       "      <td>nanograms per milliliter</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CN19SESPKOA_SC55_eDNA</th>\n",
       "      <td>12S</td>\n",
       "      <td>CE</td>\n",
       "      <td>CE6</td>\n",
       "      <td>ESP</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>5/30/19 6:17</td>\n",
       "      <td>5/30/19 7:00</td>\n",
       "      <td>25.6208</td>\n",
       "      <td>-0.0159</td>\n",
       "      <td>...</td>\n",
       "      <td>12S amplicon metabarcoding of marine metagenom...</td>\n",
       "      <td>AMPLICON</td>\n",
       "      <td>METAGENOMIC</td>\n",
       "      <td>PCR</td>\n",
       "      <td>paired</td>\n",
       "      <td>ILLUMINA</td>\n",
       "      <td>Illumina MiSeq</td>\n",
       "      <td>Seawater was filtered onto 0.22 PVDF membrane ...</td>\n",
       "      <td>fastq</td>\n",
       "      <td>nanograms per milliliter</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CN19SESPKOA_SC07_eDNA</th>\n",
       "      <td>12S</td>\n",
       "      <td>CE</td>\n",
       "      <td>CE32</td>\n",
       "      <td>ESP</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>6/6/19 7:21</td>\n",
       "      <td>6/6/19 8:04</td>\n",
       "      <td>25.8611</td>\n",
       "      <td>0.0925</td>\n",
       "      <td>...</td>\n",
       "      <td>12S amplicon metabarcoding of marine metagenom...</td>\n",
       "      <td>AMPLICON</td>\n",
       "      <td>METAGENOMIC</td>\n",
       "      <td>PCR</td>\n",
       "      <td>paired</td>\n",
       "      <td>ILLUMINA</td>\n",
       "      <td>Illumina MiSeq</td>\n",
       "      <td>Seawater was filtered onto 0.22 PVDF membrane ...</td>\n",
       "      <td>fastq</td>\n",
       "      <td>nanograms per milliliter</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CN19SESPKOA_SC06_eDNA</th>\n",
       "      <td>12S</td>\n",
       "      <td>CE</td>\n",
       "      <td>CE33</td>\n",
       "      <td>ESP</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>6/6/19 8:24</td>\n",
       "      <td>6/6/19 9:07</td>\n",
       "      <td>26.2025</td>\n",
       "      <td>0.1392</td>\n",
       "      <td>...</td>\n",
       "      <td>12S amplicon metabarcoding of marine metagenom...</td>\n",
       "      <td>AMPLICON</td>\n",
       "      <td>METAGENOMIC</td>\n",
       "      <td>PCR</td>\n",
       "      <td>paired</td>\n",
       "      <td>ILLUMINA</td>\n",
       "      <td>Illumina MiSeq</td>\n",
       "      <td>Seawater was filtered onto 0.22 PVDF membrane ...</td>\n",
       "      <td>fastq</td>\n",
       "      <td>nanograms per milliliter</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CN19SESPKOA_SC05_eDNA</th>\n",
       "      <td>12S</td>\n",
       "      <td>CE</td>\n",
       "      <td>CE34</td>\n",
       "      <td>ESP</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>6/6/19 9:24</td>\n",
       "      <td>6/6/19 10:06</td>\n",
       "      <td>26.2557</td>\n",
       "      <td>0.1402</td>\n",
       "      <td>...</td>\n",
       "      <td>12S amplicon metabarcoding of marine metagenom...</td>\n",
       "      <td>AMPLICON</td>\n",
       "      <td>METAGENOMIC</td>\n",
       "      <td>PCR</td>\n",
       "      <td>paired</td>\n",
       "      <td>ILLUMINA</td>\n",
       "      <td>Illumina MiSeq</td>\n",
       "      <td>Seawater was filtered onto 0.22 PVDF membrane ...</td>\n",
       "      <td>fastq</td>\n",
       "      <td>nanograms per milliliter</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CN19SESPKOA_SC04_eDNA</th>\n",
       "      <td>12S</td>\n",
       "      <td>CE</td>\n",
       "      <td>CE35</td>\n",
       "      <td>ESP</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>6/6/19 10:24</td>\n",
       "      <td>6/6/19 11:05</td>\n",
       "      <td>26.3305</td>\n",
       "      <td>0.1342</td>\n",
       "      <td>...</td>\n",
       "      <td>12S amplicon metabarcoding of marine metagenom...</td>\n",
       "      <td>AMPLICON</td>\n",
       "      <td>METAGENOMIC</td>\n",
       "      <td>PCR</td>\n",
       "      <td>paired</td>\n",
       "      <td>ILLUMINA</td>\n",
       "      <td>Illumina MiSeq</td>\n",
       "      <td>Seawater was filtered onto 0.22 PVDF membrane ...</td>\n",
       "      <td>fastq</td>\n",
       "      <td>nanograms per milliliter</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CN19SESPKOA_SC03_eDNA</th>\n",
       "      <td>12S</td>\n",
       "      <td>CE</td>\n",
       "      <td>CE36</td>\n",
       "      <td>ESP</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>6/6/19 11:25</td>\n",
       "      <td>6/6/19 12:09</td>\n",
       "      <td>26.4586</td>\n",
       "      <td>0.0992</td>\n",
       "      <td>...</td>\n",
       "      <td>12S amplicon metabarcoding of marine metagenom...</td>\n",
       "      <td>AMPLICON</td>\n",
       "      <td>METAGENOMIC</td>\n",
       "      <td>PCR</td>\n",
       "      <td>paired</td>\n",
       "      <td>ILLUMINA</td>\n",
       "      <td>Illumina MiSeq</td>\n",
       "      <td>Seawater was filtered onto 0.22 PVDF membrane ...</td>\n",
       "      <td>fastq</td>\n",
       "      <td>nanograms per milliliter</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>245 rows × 31 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                      target_gene PlateID library_ID Sampling_method  \\\n",
       "sample_name                                                            \n",
       "CN19SESPMV1_SC58_eDNA         12S      CE       CE40             ESP   \n",
       "CN19SESPKOA_SC58_eDNA         12S      CE        CE3             ESP   \n",
       "CN19SESPKOA_SC57_eDNA         12S      CE        CE4             ESP   \n",
       "CN19SESPKOA_SC56_eDNA         12S      CE        CE5             ESP   \n",
       "CN19SESPKOA_SC55_eDNA         12S      CE        CE6             ESP   \n",
       "...                           ...     ...        ...             ...   \n",
       "CN19SESPKOA_SC07_eDNA         12S      CE       CE32             ESP   \n",
       "CN19SESPKOA_SC06_eDNA         12S      CE       CE33             ESP   \n",
       "CN19SESPKOA_SC05_eDNA         12S      CE       CE34             ESP   \n",
       "CN19SESPKOA_SC04_eDNA         12S      CE       CE35             ESP   \n",
       "CN19SESPKOA_SC03_eDNA         12S      CE       CE36             ESP   \n",
       "\n",
       "                       replicate  SAMPLING_rdepth      start_GMT  \\\n",
       "sample_name                                                        \n",
       "CN19SESPMV1_SC58_eDNA        NaN              NaN   5/29/19 9:42   \n",
       "CN19SESPKOA_SC58_eDNA        NaN              NaN  5/29/19 19:32   \n",
       "CN19SESPKOA_SC57_eDNA        NaN              NaN  5/29/19 20:47   \n",
       "CN19SESPKOA_SC56_eDNA        NaN              NaN  5/29/19 21:50   \n",
       "CN19SESPKOA_SC55_eDNA        NaN              NaN   5/30/19 6:17   \n",
       "...                          ...              ...            ...   \n",
       "CN19SESPKOA_SC07_eDNA        NaN              NaN    6/6/19 7:21   \n",
       "CN19SESPKOA_SC06_eDNA        NaN              NaN    6/6/19 8:24   \n",
       "CN19SESPKOA_SC05_eDNA        NaN              NaN    6/6/19 9:24   \n",
       "CN19SESPKOA_SC04_eDNA        NaN              NaN   6/6/19 10:24   \n",
       "CN19SESPKOA_SC03_eDNA        NaN              NaN   6/6/19 11:25   \n",
       "\n",
       "                             end_GMT   sigmat   spice  ...  \\\n",
       "sample_name                                            ...   \n",
       "CN19SESPMV1_SC58_eDNA  5/29/19 10:07  25.0274  0.2338  ...   \n",
       "CN19SESPKOA_SC58_eDNA  5/29/19 20:18  25.4315  0.0592  ...   \n",
       "CN19SESPKOA_SC57_eDNA  5/29/19 21:28  26.5216  0.0742  ...   \n",
       "CN19SESPKOA_SC56_eDNA  5/29/19 22:33  26.6542  0.0243  ...   \n",
       "CN19SESPKOA_SC55_eDNA   5/30/19 7:00  25.6208 -0.0159  ...   \n",
       "...                              ...      ...     ...  ...   \n",
       "CN19SESPKOA_SC07_eDNA    6/6/19 8:04  25.8611  0.0925  ...   \n",
       "CN19SESPKOA_SC06_eDNA    6/6/19 9:07  26.2025  0.1392  ...   \n",
       "CN19SESPKOA_SC05_eDNA   6/6/19 10:06  26.2557  0.1402  ...   \n",
       "CN19SESPKOA_SC04_eDNA   6/6/19 11:05  26.3305  0.1342  ...   \n",
       "CN19SESPKOA_SC03_eDNA   6/6/19 12:09  26.4586  0.0992  ...   \n",
       "\n",
       "                                                                   title  \\\n",
       "sample_name                                                                \n",
       "CN19SESPMV1_SC58_eDNA  12S amplicon metabarcoding of marine metagenom...   \n",
       "CN19SESPKOA_SC58_eDNA  12S amplicon metabarcoding of marine metagenom...   \n",
       "CN19SESPKOA_SC57_eDNA  12S amplicon metabarcoding of marine metagenom...   \n",
       "CN19SESPKOA_SC56_eDNA  12S amplicon metabarcoding of marine metagenom...   \n",
       "CN19SESPKOA_SC55_eDNA  12S amplicon metabarcoding of marine metagenom...   \n",
       "...                                                                  ...   \n",
       "CN19SESPKOA_SC07_eDNA  12S amplicon metabarcoding of marine metagenom...   \n",
       "CN19SESPKOA_SC06_eDNA  12S amplicon metabarcoding of marine metagenom...   \n",
       "CN19SESPKOA_SC05_eDNA  12S amplicon metabarcoding of marine metagenom...   \n",
       "CN19SESPKOA_SC04_eDNA  12S amplicon metabarcoding of marine metagenom...   \n",
       "CN19SESPKOA_SC03_eDNA  12S amplicon metabarcoding of marine metagenom...   \n",
       "\n",
       "                       library_strategy  library_source  library_selection  \\\n",
       "sample_name                                                                  \n",
       "CN19SESPMV1_SC58_eDNA          AMPLICON     METAGENOMIC                PCR   \n",
       "CN19SESPKOA_SC58_eDNA          AMPLICON     METAGENOMIC                PCR   \n",
       "CN19SESPKOA_SC57_eDNA          AMPLICON     METAGENOMIC                PCR   \n",
       "CN19SESPKOA_SC56_eDNA          AMPLICON     METAGENOMIC                PCR   \n",
       "CN19SESPKOA_SC55_eDNA          AMPLICON     METAGENOMIC                PCR   \n",
       "...                                 ...             ...                ...   \n",
       "CN19SESPKOA_SC07_eDNA          AMPLICON     METAGENOMIC                PCR   \n",
       "CN19SESPKOA_SC06_eDNA          AMPLICON     METAGENOMIC                PCR   \n",
       "CN19SESPKOA_SC05_eDNA          AMPLICON     METAGENOMIC                PCR   \n",
       "CN19SESPKOA_SC04_eDNA          AMPLICON     METAGENOMIC                PCR   \n",
       "CN19SESPKOA_SC03_eDNA          AMPLICON     METAGENOMIC                PCR   \n",
       "\n",
       "                      library_layout  platform  instrument_model  \\\n",
       "sample_name                                                        \n",
       "CN19SESPMV1_SC58_eDNA         paired  ILLUMINA    Illumina MiSeq   \n",
       "CN19SESPKOA_SC58_eDNA         paired  ILLUMINA    Illumina MiSeq   \n",
       "CN19SESPKOA_SC57_eDNA         paired  ILLUMINA    Illumina MiSeq   \n",
       "CN19SESPKOA_SC56_eDNA         paired  ILLUMINA    Illumina MiSeq   \n",
       "CN19SESPKOA_SC55_eDNA         paired  ILLUMINA    Illumina MiSeq   \n",
       "...                              ...       ...               ...   \n",
       "CN19SESPKOA_SC07_eDNA         paired  ILLUMINA    Illumina MiSeq   \n",
       "CN19SESPKOA_SC06_eDNA         paired  ILLUMINA    Illumina MiSeq   \n",
       "CN19SESPKOA_SC05_eDNA         paired  ILLUMINA    Illumina MiSeq   \n",
       "CN19SESPKOA_SC04_eDNA         paired  ILLUMINA    Illumina MiSeq   \n",
       "CN19SESPKOA_SC03_eDNA         paired  ILLUMINA    Illumina MiSeq   \n",
       "\n",
       "                                                      design_description  \\\n",
       "sample_name                                                                \n",
       "CN19SESPMV1_SC58_eDNA  Seawater was filtered onto 0.22 PVDF membrane ...   \n",
       "CN19SESPKOA_SC58_eDNA  Seawater was filtered onto 0.22 PVDF membrane ...   \n",
       "CN19SESPKOA_SC57_eDNA  Seawater was filtered onto 0.22 PVDF membrane ...   \n",
       "CN19SESPKOA_SC56_eDNA  Seawater was filtered onto 0.22 PVDF membrane ...   \n",
       "CN19SESPKOA_SC55_eDNA  Seawater was filtered onto 0.22 PVDF membrane ...   \n",
       "...                                                                  ...   \n",
       "CN19SESPKOA_SC07_eDNA  Seawater was filtered onto 0.22 PVDF membrane ...   \n",
       "CN19SESPKOA_SC06_eDNA  Seawater was filtered onto 0.22 PVDF membrane ...   \n",
       "CN19SESPKOA_SC05_eDNA  Seawater was filtered onto 0.22 PVDF membrane ...   \n",
       "CN19SESPKOA_SC04_eDNA  Seawater was filtered onto 0.22 PVDF membrane ...   \n",
       "CN19SESPKOA_SC03_eDNA  Seawater was filtered onto 0.22 PVDF membrane ...   \n",
       "\n",
       "                      filetype         concentrationUnit  \n",
       "sample_name                                               \n",
       "CN19SESPMV1_SC58_eDNA    fastq  nanograms per milliliter  \n",
       "CN19SESPKOA_SC58_eDNA    fastq  nanograms per milliliter  \n",
       "CN19SESPKOA_SC57_eDNA    fastq  nanograms per milliliter  \n",
       "CN19SESPKOA_SC56_eDNA    fastq  nanograms per milliliter  \n",
       "CN19SESPKOA_SC55_eDNA    fastq  nanograms per milliliter  \n",
       "...                        ...                       ...  \n",
       "CN19SESPKOA_SC07_eDNA    fastq  nanograms per milliliter  \n",
       "CN19SESPKOA_SC06_eDNA    fastq  nanograms per milliliter  \n",
       "CN19SESPKOA_SC05_eDNA    fastq  nanograms per milliliter  \n",
       "CN19SESPKOA_SC04_eDNA    fastq  nanograms per milliliter  \n",
       "CN19SESPKOA_SC03_eDNA    fastq  nanograms per milliliter  \n",
       "\n",
       "[245 rows x 31 columns]"
      ]
     },
     "execution_count": 95,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df = pd.concat([metas[0], extract], axis=1)\n",
    "df = pd.concat([df, meta_all[['R1', 'R2']]], axis=1)\n",
    "#df = df.loc[df['water_filt_mL'].isna()==True]\n",
    "# drop columns that are only for internal use\n",
    "cols = ['ESP_name', 'diel', 'month', 'day', 'hour', 'time_label', 'local_time', 'SC', 'ESP', 'Plates', 'Markers', 'Status', 'Dewar_name']\n",
    "df = df.drop(cols, axis=1)\n",
    "\n",
    "\n",
    "\n",
    "# convert names to NOAA standard terms\n",
    "df = df.rename(columns={'SAMPLING_cruise':'cruise_id',\n",
    "                       'FilterID':'source_material_id',\n",
    "                        'SAMPLING_station_number':'station',\n",
    "                        'SAMPLING_bottle':'ctd_bottle_no',\n",
    "                        'water_filt_mL':'samp_vol_we_dna_ext',\n",
    "                        'SAMPLING_platform':'samp_collect_device',\n",
    "                        'SAMPLING_station':'locationID',\n",
    "                        'DNA_concentration_ng_uL':'dna_conc',\n",
    "                        'concentrationUnit' : 'measurement_method',\n",
    "                        'library' : 'library_ID',\n",
    "                        'R1': 'filename',\n",
    "                        'R2': 'filename2'\n",
    "                       })\n",
    "\n",
    "\n",
    "\n",
    "# Standard Columns to Add\n",
    "# df['organism'] = 'seawater metagenome'  # only if environmental sample\n",
    "df['organism'] = ''\n",
    "df.loc[df['sample_type']=='environmental', 'organism'] = 'seawater metagenome'\n",
    "# term 'control sample' can be used for negative controls: (https://www.ddbj.nig.ac.jp/biosample/submission-e.html)\n",
    "df.loc[df['sample_type'].isin(['negative','MSU_control'])==True, 'organism'] = 'control sample'\n",
    "# for artificial/mock communities, term is synthetic metagenome\n",
    "df.loc[df.index.str.contains('ArtC|Art_C', case=False), 'organism'] = 'synthetic metagenome'\n",
    "# ESP positive controls are seawater\n",
    "df.loc[(df['sample_type']=='positive') & (df['Sampling_method']=='ESP'), 'organism'] = 'seawater metagenome'\n",
    "\n",
    "print(df['organism'].unique())\n",
    "# If sample is from seawater the following applies:\n",
    "df.loc[df['organism']=='seawater metagenome','env_broad_scale' ] = 'marine biome [ENVO:00000447]'\n",
    "df.loc[df['organism']=='seawater metagenome','env_local_scale' ] = 'marine mesopelagic zone [ENVO:00000213]'\n",
    "df.loc[df['organism']=='seawater metagenome','env_medium' ] = 'sea water [ENVO:00002149]'\n",
    "df.loc[df['organism']=='seawater metagenome','geo_loc_name' ] = 'USA: Monterey Bay'\n",
    "df.loc[df['organism']=='seawater metagenome','size_frac' ] = '0.22 micrometer'\n",
    "\n",
    "df['collection_method'] = df['Sampling_method']\n",
    "df['collection_method'] = df['collection_method'].replace('CTD', 'CTD rosette')\n",
    "df['collection_method'] = df['collection_method'].replace('ESP', 'Environmental Sample Processor onboard long-range autonomous underwater vehicle (LRAUV-ESP)')\n",
    "print(df['collection_method'].unique())\n",
    "# date in UTC\n",
    "df['collection_date'] = df['eventDate']\n",
    "# need to be in these formats:\n",
    "# \"DD-Mmm-YYYY\", \"Mmm-YYYY\", \"YYYY\" or ISO 8601 standard \"YYYY-mm-dd\", \"YYYY-mm\", \"YYYY-mm-ddThh:mm:ss\";\n",
    "df['collection_date'] = pd.to_datetime(df['collection_date'])\n",
    "df['collection_date'] = df['collection_date'].dt.strftime('%Y-%m-%dT%H:%M:%S')\n",
    "df['collection_date'] = df['collection_date'] +'Z'\n",
    "# join lat lon information\n",
    "# since we are adding 'W' we can drop the negative before the longitude value\n",
    "df['decimalLongitude_str'] = -df['decimalLongitude']\n",
    "df['lat_lon'] = df['decimalLatitude'].astype(str) +' N ' + df['decimalLongitude_str'].astype(str) + ' W'\n",
    "df['lat_lon'] = df['lat_lon'].str.replace('nan N nan W', '')\n",
    "# study level metadata\n",
    "df['project_id'] = 'CN19S'\n",
    "df['project_name'] = 'eDNA from Controlled Agile and Novel Observing Network (CANON) cruise in spring of 2019 (CN19S)'\n",
    "df['project_id_external'] = 'MBARI-BOG-CN19S'\n",
    "df['project_contact'] = 'Kathleen Pitz, kpitz@mbari.org, https://orcid.org/0000-0002-4931-8592'\n",
    "df['project_description'] = ''\n",
    "df['type'] = 'Occurence'\n",
    "df['license'] = 'CC-BY 4.0'\n",
    "# Amplicon Prep Data Columns (NOAA - becomes SRA metadata)\n",
    "df['title'] = '12S amplicon metabarcoding of marine metagenome: Monterey Bay, CA (USA)'\n",
    "df['library_strategy'] = 'AMPLICON'\n",
    "df['library_source'] = 'METAGENOMIC'\n",
    "df['library_selection'] = 'PCR'\n",
    "# library_layout - 'lib_layout' in NOAA sheet\n",
    "df['library_layout'] = 'paired'\n",
    "df['platform'] = 'ILLUMINA'\n",
    "df['instrument_model'] = 'Illumina MiSeq'\n",
    "df['design_description'] = 'Seawater was filtered onto 0.22 PVDF membrane filters through shipboard and vehicle sampling'\n",
    "df['filetype'] = 'fastq'\n",
    "# design description supposed to be a mini 'methods' section:\n",
    "df.loc[df['PlateID'].isin(['JJ', 'RR']), 'design_description'] = 'Seawater was filtered onto 0.22 PVDF membrane filters through shipboard and vehicle sampling. \\\n",
    "DNA was extracted using the Qiagen DNeasy Blood and Tissue Kit. A two-step PCR approach was used, targeting mitochondrial 12S with primers MiFish-U-F (5-GTCGGTAAAACTCGTGCCAGC-3) \\\n",
    "and MiFish-U-R (5-CATAGTGGGGTATCTAATCCCAGTTTG-3). Fluidigm tail sequences CS1 (5-ACACTGACGACATGGTTCTACA-3) and CS2 (5-TACGGTAGCAGAGACTTGGTCT-3) were added to the 5-prime ends of \\\n",
    "the forward and reverse primer sequences. Secondary amplification and next-generation sequencing were performed at Michigan State University where Illumina compatible adapters with \\\n",
    "barcodes targeted CS1/CS2 ends of the primery PCR products. Pooled product was sequenced in a 2x250bp paired end format on an Illumina MiSeq.'\n",
    "\n",
    "df.loc[df['PlateID'].isin(['CE', 'BT']), 'design_description'] = 'Seawater was filtered onto 0.22 PVDF membrane filters through shipboard and vehicle sampling. \\\n",
    "DNA was extracted using the Qiagen DNeasy Blood and Tissue Kit. A two-step PCR approach was used, targeting mitochondrial 12S with primers MiFish-U-F (5-GTCGGTAAAACTCGTGCCAGC-3) \\\n",
    "and MiFish-U-R (5-CATAGTGGGGTATCTAATCCCAGTTTG-3). Fluidigm tail sequences CS1 (5-ACACTGACGACATGGTTCTACA-3) and CS2 (5-TACGGTAGCAGAGACTTGGTCT-3) were added to the 5-prime ends of \\\n",
    "the forward and reverse primer sequences. Secondary amplification and next-generation sequencing were performed at Michigan State University where Illumina compatible adapters with \\\n",
    "barcodes targeted CS1/CS2 ends of the primery PCR products. Pooled product was size selected with a Pippin HT to select for the vertebrate/fish band (~350bp) and remove co-amplified \\\n",
    "bacteria (~435bp). It was then sequenced in a 2x250bp paired end format on an Illumina MiSeq.'\n",
    "\n",
    "print(df['design_description'].unique())\n",
    "\n",
    "# we use sample_type differently - if 'environmental', should be 'seawater'\n",
    "df['sample_type'] = df['sample_type'].str.replace('environmental', 'seawater')\n",
    "\n",
    "# mandatory columns, followed by optional MixS MIMARKS columns; index is sample_name (mandatory); export as tab deliminated\n",
    "Biosample_cols = ['organism', 'collection_date', 'depth', 'env_broad_scale', 'env_local_scale', 'env_medium', 'geo_loc_name', 'lat_lon',\n",
    "                 'chlorophyll', 'density', 'diss_oxygen', 'fluor', 'nitrate', 'pressure', 'salinity', 'samp_collect_device',\n",
    "                  'samp_vol_we_dna_ext', 'source_material_id', 'temp']\n",
    "# additional NOAA columns in water_sample_data tab - can include\n",
    "NOAA_water_sample_data = ['cruise_id', 'station', 'locationID', 'ctd_bottle_no', 'decimalLongitude', 'decimalLatitude', 'sample_type',\n",
    "                         'dna_conc','measurement_method','minimumDepthInMeters', 'maximumDepthInMeters', 'collection_method']\n",
    "# Used to generate BioProject\n",
    "NOAA_study_data = ['project_id', 'project_name','project_id_external','project_contact', 'project_description', 'type','license'  ]\n",
    "# Will need to be submitted as SRA metadata\n",
    "NOAA_amplicon_prep_data = ['library_ID', 'title', 'library_strategy', 'library_source', 'library_selection', 'library_layout',\n",
    "                            'platform', 'instrument_model', 'design_description', 'filetype', 'filename', 'filename2']\n",
    "#df = df[Biosample_cols]\n",
    "\n",
    "\n",
    "\n",
    "# need to add units to measurements\n",
    "# density is currently sigma-t (kg/m^3 - 1000)\n",
    "df['density'] = df['density'] + 1000\n",
    "cols = ['density', 'chlorophyll', 'depth', 'diss_oxygen', 'fluor', 'nitrate', 'pressure', 'salinity', 'samp_vol_we_dna_ext', 'temp', 'dna_conc']\n",
    "add_units = ['kilogram per cubic meter', 'milligram per cubic meter', 'meter', 'milliliter per liter', 'volts',\n",
    "             'micromole per liter', 'decibar', 'parts per thousand', 'milliliter', 'degree Celsius', '']\n",
    "\n",
    "for i, unit in zip(cols,add_units):\n",
    "    print(i)\n",
    "    print(unit)\n",
    "    #df.loc[df[i].isna()==False,df[i]] = df[i].astype(str) + ' ' + unit\n",
    "    df[i] = df[i].astype('string') + ' ' + unit\n",
    "    # if was nan beforehand then don't want a field now\n",
    "\n",
    "# samp_collect_device has controlled vocabulary need to change\n",
    "# concentrationUnit - dna_conc unit\n",
    "df['concentrationUnit'] = 'nanograms per milliliter'\n",
    "\n",
    "# For the biosample submission, remove control samples and just include environmental samples\n",
    "# only environmental samples have 'depth'\n",
    "df = df.loc[df['depth'].isna()==False]\n",
    "\n",
    "# change sample name to be the filter name, not the sequenced sample id (remove the plate ID)\n",
    "df = df.reset_index()\n",
    "\n",
    "df['sample_name'] = df['sample_name'].str.split('_').str[:-1]\n",
    "df['sample_name'] = df['sample_name'].str.join('_')\n",
    "print(len(df['sample_name'].unique()))\n",
    "df.set_index('sample_name', inplace=True)\n",
    "\n",
    "# remove samples not included because they had fewer than 500 reads:\n",
    "df = df.loc[df['samp_vol_we_dna_ext'].isna()==False]\n",
    "print(len(df.index))\n",
    "\n",
    "file = '/Users/kpitz/Documents/biosample_metadata.csv'\n",
    "df[Biosample_cols+NOAA_water_sample_data].to_csv(file, sep='\\t')\n",
    "\n",
    "# SRA: If you are submitting metagenomic and/or metatranscriptomic data sets, sequence data should be split by each sample barcode, for individual data files\n",
    "\n",
    "file = '/Users/kpitz/Documents/sra_metadata.csv'\n",
    "df[NOAA_amplicon_prep_data].to_csv(file, sep='\\t')\n",
    "\n",
    "\n",
    "# generate script (.sh file) to copy fastq files to the same folder to make submission easier?\n",
    "# /Volumes/MBON/raw_sequence_data/12S/CE_20220629_12S_PE250/CE1/CN19SESPKoa_SC60_eDNA_S1_L001_R1_001.fastq.gz\n",
    "# cp [original_location + filename] [new_location + filename]\n",
    "#plate_folders = ['CE_20220629_12S_PE250', 'BT_20220425_12S_PE250', 'JJ_20191127_12S_PE250', 'RR_20200702_12S_PE250']\n",
    "#plate_ids = ['CE', 'BT', 'JJ', 'RR']\n",
    "original_location = '/Volumes/MBON/raw_sequence_data/12S/'\n",
    "\n",
    "new_location = '/Users/kpitz/Projects/SRA_submission/DVM_CN19S/'\n",
    "\n",
    "R1s = df['filename'].tolist()\n",
    "R2s = df['filename2'].tolist()\n",
    "plates = df['PlateID'].tolist()\n",
    "lib_ids = df['library_ID'].tolist()\n",
    "print(df['PlateID'].unique())\n",
    "outfile = '/Users/kpitz/Documents/copy_fastq_files.sh'\n",
    "\n",
    "##Write File\n",
    "resultsFile = open(outfile, \"w\") #open resultsfile\n",
    "pfolder=[]\n",
    "for i in range(len(R1s)):\n",
    "    if plates[i] == 'CE':\n",
    "        pfolder = 'CE_20220629_12S_PE250'\n",
    "    elif plates[i] == 'BT':\n",
    "        pfolder = 'BT_20220425_12S_PE250'\n",
    "    elif plates[i] == 'JJ':\n",
    "        pfolder = 'JJ_20191127_12S_PE250'    \n",
    "    elif plates[i] == 'RR':\n",
    "        pfolder = 'RR_20200702_12S_PE250'    \n",
    "    else:\n",
    "        print('ERROR: unknown plate ', plates[i])\n",
    "        break\n",
    "    resultsFile.write('cp '+ original_location + pfolder+'/'+lib_ids[i]+ '/'+R1s[i]+' '+ new_location+R1s[i]+'\\n')\n",
    "    resultsFile.write('cp '+ original_location + pfolder+'/'+lib_ids[i]+ '/'+R2s[i]+' '+ new_location+R2s[i]+'\\n')\n",
    "resultsFile.close()\n",
    "\n",
    "print('Done!')\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "# look at what's left:\n",
    "df = df.drop(Biosample_cols, axis=1)\n",
    "df = df.drop(NOAA_water_sample_data, axis=1)\n",
    "df = df.drop(NOAA_study_data, axis=1)\n",
    "\n",
    "print(df['Sampling_method'].unique())\n",
    "print(list(df))\n",
    "\n",
    "df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "4624e2b2-9d3f-45ea-afa4-610147bf48d9",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "34cf5560-7ab7-4e82-a3e4-4b83c3d82f00",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.11.6"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
